api/tests/mocha/services/report/smsparser.spec.js

```diff
@@ -97,10 +97,71 @@ describe('sms parser', () => {
     chai.expect(smsparser.getFormCode('द CDT33')).to.equal('द');
   });
 
+  it('accepts phone number with extension', () => {
+    const doc = { message: 'NP 20 +9779841202020' };
+    const def = definitions.forms.NP;
+    sinon.stub(config, 'getAll').returns({
+      default_country_code: 977,
+      phone_validation: 'full'
+    });
+    const data = smsparser.parse(def, doc);
+    chai.expect(data.phone_number).to.equal('+9779841202020');
+  });
+
+  it('accepts correct phone number without extension', () => {
+    const doc = { message: 'NP 20 9841202020' };
+    const def = definitions.forms.NP;
+    sinon.stub(config, 'getAll').returns({
+      default_country_code: 977,
+      phone_validation: 'full'
+    });
+    const data = smsparser.parse(def, doc);
+    chai.expect(data.phone_number).to.equal('+9779841202020');
+  });
+
+  it('returns null if phone number is invalid for the region', () => {
+    const doc = { message: 'NP 20 +97712312 Prajwol' };
+    const def = definitions.forms.NP;
+    sinon.stub(config, 'getAll').returns({
+      default_country_code: 977,
+      phone_validation: 'full'
+    });
+    const data = smsparser.parse(def, doc);
+    chai.expect(data.phone_number).to.equal(null);
+  });
+
+  it('returns null if phone number is invalid for default region', () => {
+    const doc = { message: 'NP 20 8750660880 Prajwol' };
+    const def = definitions.forms.NP;
+    sinon.stub(config, 'getAll').returns({
+      default_country_code: 977,
+      phone_validation: 'full'
+    });
+    const data = smsparser.parse(def, doc);
+    chai.expect(data.phone_number).to.equal(null);
+  });
+
+  //India , Kenya, Tanzania Phone is accepted as contact info in Nepal region.
+  //Just incase we make cross region/borders tool
+  [['NP 20 +918750660880 Prajwol', '+918750660880'],
+    ['NP 20 +254773087889 Prajwol', '+254773087889'],
+    ['NP 20 +255712262987 Prajwol', '+255712262987']].forEach(phoneNumerWithParsed => {
+    it('returns parsed number if valid phone of another the region', () => {
```

Comment:
This will list three unit tests with identical names. If one of them fails it won't be immediately obvious which one. Let's include the message in the test name:

```suggestion
    it(`returns parsed number if valid phone of another the region: ${phoneNumerWithParsed[0]}`, () => {
```